src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -173,7 +173,14 @@ protected override void Dispose(bool disposing)
                         var attr = File.GetAttributes(copiedFile);
                         if (!attr.HasFlag(FileAttributes.Directory))
                         {
-                            File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            try
+                            {
+                                File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            }
+                            catch (ArgumentOutOfRangeException)
```

Comment:
Here's better info: The zip entry's LastWriteTime is actually not set, and calling entry.LastWriteTime.UtcDateTime will return a zero DateTime (minimum).

On Windows, calling File.SetLastWriteTimeUtc with a zero DateTime works and uses 1980/1/1 instead. On Mono, it throws ArgumentOutOfRangeException. That is a valid exception to throw for an invalid time, .Net on Windows is just being nice and picking some other time.

So now I don't think this is a bug in Mono, it's doing something sort of reasonable.